pkg/sql/opt/optbuilder/delete.go

```diff
@@ -74,8 +74,15 @@ func (b *Builder) buildDelete(del *tree.Delete, inScope *scope) (outScope *scope
 // buildDelete constructs a Delete operator, possibly wrapped by a Project
 // operator that corresponds to the given RETURNING clause.
 func (mb *mutationBuilder) buildDelete(returning tree.ReturningExprs) {
+	// Disambiguate names so that references in any expressions, such as a
+	// partial index predicate, refer to the correct columns.
+	mb.disambiguateColumns()
```

Comment:
Is this necessary in this case? I _think_ so, but I'm not fully certain about the purpose of `disambiguateColumns`.